woocommerce/woocommerce-ios
# [Tablet Orders] Create order: Address design review items

Closes: #12155

## Description
This PR tackles the design review items logged for "Create Order", that is:

<img width="361" alt="Screenshot 2024-03-01 at 14 23 23" src="https://github.com/woocommerce/woocommerce-ios/assets/3812076/5c8d3834-2bdb-4f17-87a9-83c22d1eacce">

* Adds missing divider lines wrapping the `AddOrderComponentsSection` ("+ Add Shipping", "+ Add Coupon", ... )
* Makes all tappable buttons/rows 56pt height
* Reduces the existing space above and below button groups from 16pt to 8pt
* Adds a 8pt space to the top of the view

I have applied these without a conditional  `sideBySideViewForOrderForm` feature flag usage, since are design updates that are not coming only with this project but should affect the Order Form view across all its instances (split view, modal on modal, or plain iPhone). Happy to add additional logic if you think we should keep the previous UI if the flag is off.

## Testing instructions
1. With the feature flag off, observe that changes are applied to modal-on-modal, and iPhone
![simulator_screenshot_446BA2A5-074C-47BA-8476-A1A5ED79DC30](https://github.com/woocommerce/woocommerce-ios/assets/3812076/9e0267e8-1d1e-4d6a-b363-c6688ed2c0c1)

2. With the feature flag on, observe that changes are applied to side-by-side, and iPhone
![20240301-changes-applied-side-by-side](https://github.com/woocommerce/woocommerce-ios/assets/3812076/8813b98c-3c6d-4059-83e9-0a229852dc27)

Adding products, shipping, etc, ... should behave normally in all cases.
